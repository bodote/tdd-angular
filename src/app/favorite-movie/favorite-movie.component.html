<p>Movie: {{movieName}} 
<label>delete an item: </label>
<button (click)="deleteMovie(movieName)">delete</button>
</p>