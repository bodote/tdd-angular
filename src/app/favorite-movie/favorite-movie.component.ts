import { Component, Input, OnInit, Output,EventEmitter } from '@angular/core';


@Component({
  selector: 'favorite-movie',
  templateUrl: './favorite-movie.component.html',
  styleUrls: ['./favorite-movie.component.css']
})
export class FavoriteMovieComponent implements OnInit {
@Input() movieName: string
@Output() deleteEmitter : EventEmitter<string>
  constructor() { 
    this.deleteEmitter = new EventEmitter<string>()
  }

  ngOnInit(): void {
    
  }
  deleteMovie(movie:string):void{
    this.deleteEmitter.emit(movie)
  }

}
