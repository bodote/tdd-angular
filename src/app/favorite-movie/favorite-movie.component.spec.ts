import { ComponentFixture, TestBed } from '@angular/core/testing';

import { FavoriteMovieComponent } from './favorite-movie.component';


describe('FavoriteMovieComponent', () => {
  let component: FavoriteMovieComponent;
  let fixture: ComponentFixture<FavoriteMovieComponent>;;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ FavoriteMovieComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(FavoriteMovieComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should contain label with text',()=>{
    const nativeElement = fixture.nativeElement;
    expect(nativeElement.querySelector('label').textContent).toContain('delete');
    const expFix = expect(fixture)
    expFix.toMatchSnapshot()
  
  })
});
