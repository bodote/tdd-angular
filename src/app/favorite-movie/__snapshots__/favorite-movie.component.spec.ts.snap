// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteMovieComponent should contain label with text 1`] = `
<favorite-movie
  deleteEmitter={[Function EventEmitter_]}
>
  <p>
    Movie:  
    <label>
      delete an item: 
    </label>
    <button>
      delete
    </button>
  </p>
</favorite-movie>
`;
